src/Game/Systems/PlayerSpawnSystem.cpp

```diff
@@ -230,6 +241,14 @@ bool PlayerSpawnSystem::OnPlayerDeath(Events::PlayerDeath& e)
     SpawnRequest req;
     req.PlayerID = m_PlayerIDs.at(e.Player.ID);
     req.Team = cTeam["Team"];
+    // TODO: Better than temp class state code.
+    if (e.Player.HasComponent("DashAbility")) {
+        req.Class = 1;      // Assault
+    } else if (e.Player.HasComponent("SprintAbility")) {
+        req.Class = 3;      // Sniper
+    } else {
+        req.Class = 2;      // Defender
```

Comment:
What if player has no ability? Then it becomes Defender, is this intended? If not ShieldAbility could be checked for.